nengo/config.py

```diff
@@ -84,6 +84,10 @@ def __repr__(self):
         return "<%s[%s]{%s}>" % (type(self).__name__,
                                  self._configures.__name__, ", ".join(params))
 
+    def __setstate__(self, state):
+        for k, v in state.items():
+            setattr(self, k, v)
```

Comment:
So this is was happening: `pickle` checks whether `ClassParams` defines it's own `__setstate__` by calling `__getattribute__`. Now, if we define a `__setstate__` with default behaviour, it will find `__setstate__` and return it. But if we don't, it will delegate the request to `__getattr__` which is overwritten in `ClassParams` and leads to an infinite loop because it calls `get_param` which uses `getattr` (not sure why `self._configures` seems to end up as self).

So an alternate solution to defining `__setattr__` would be to check whether the `key` in `__getattr__` is `__setstate__` or starts with underscores or something similar and just invoke the super `__getattr__` in that case.

It also seems slightly weird to me that `get_param` and `__getattr__` invoke each other ...